cmd/kpod/history.go

```diff
@@ -0,0 +1,267 @@
+		"or user sepcified format, and can be truncated."
+	historyCommand = cli.Command{
+		Name:        "history",
+		Usage:       "Show history of a specified image",
+		Description: historyDescription,
+		Flags:       historyFlags,
+		Action:      historyCmd,
+		ArgsUsage:   "",
+	}
+)
+
+func historyCmd(c *cli.Context) error {
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	human := true
+	if c.IsSet("human") {
+		human = c.Bool("human")
+	}
+	noTruncate := false
+	if c.IsSet("no-trunc") {
+		noTruncate = c.Bool("no-trunc")
+	}
+	quiet := false
+	if c.IsSet("quiet") {
+		quiet = c.Bool("quiet")
+	}
+	format := ""
+	hasTemplate := false
+	if c.IsSet("format") {
+		format = c.String("format")
+		hasTemplate = true
+	}
+
+	args := c.Args()
+	if len(args) == 0 {
+		logrus.Errorf("an image name must be specified")
+		return nil
+	}
+	if len(args) > 1 {
+		logrus.Errorf("Kpod history takes at most 1 argument")
+		return nil
+	}
+	imgName := args[0]
+
+	opts := historyOptions{
+		image:   imgName,
+		human:   human,
+		noTrunc: noTruncate,
+		quiet:   quiet,
+		format:  format,
+	}
+
+	if !hasTemplate && !quiet {
+		outputHeader(opts)
+	}
+
+	return outputHistory(store, opts)
+}
+
+// outputHeader outputs the heading
+func outputHeader(opts historyOptions) {
+	if !opts.noTrunc {
+		fmt.Printf("%-12s\t\t%-16s\t\t%-42s\t\t", "IMAGE", "CREATED", "CREATED BY")
+		fmt.Printf("%-16s\t\t%s\n", "SIZE", "COMMENT")
+	} else {
+		fmt.Printf("%-64s\t%-15s\t%-60s\t", "IMAGE", "CREATED", "CREATED BY")
+		fmt.Printf("%-16s\t%s\n", "SIZE", "COMMENT")
+	}
+}
+
+// outputString outputs the information in historyOutputParams
+func outputString(opts historyOptions, params historyOutputParams) {
+	if !opts.noTrunc {
+		fmt.Printf("%-12.12s\t\t%-16s\t\t%-42.42s\t\t", params.ID, params.Created, params.CreatedBy)
+		fmt.Printf("%-16s\t\t%s\n", params.Size, params.Comment)
+	} else {
+		fmt.Printf("%-64s\t%-15s\t%-60s\t", params.ID, params.Created, params.CreatedBy)
+		fmt.Printf("%-16s\t%s\n\n", params.Size, params.Comment)
+	}
+}
+
+// outputWithTemplate is called when --format is given a template
+func outputWithTemplate(format string, params historyOutputParams) error {
+	templ, err := template.New("history").Parse(format)
+	if err != nil {
+		return errors.Wrapf(err, "Template parsing error")
+	}
+
+	err = templ.Execute(os.Stdout, params)
+	if err != nil {
+		return err
+	}
+	fmt.Println()
+	return nil
+}
+
+// outputTime displays the time stamp in "2017-06-20T20:24:10Z" format
+func outputTime(tm *time.Time) string {
+	newTime := tm.Format(time.RFC3339)
+	return newTime
+}
+
+// outputHumanTime displays the time elapsed since creation
+func outputHumanTime(tm *time.Time) string {
+	duration := time.Since(*tm)
+	return units.HumanDuration(duration)
+}
+
+// outputHistory retrieves the history of the image and prints it
+// based on the cli flag values
+func outputHistory(store storage.Store, opts historyOptions) error {
```

Comment:
Modify this function to create/return a json file and have this file either output directly or converted to the human print.